mne/io/pick.py

```diff
@@ -31,54 +71,17 @@ def channel_type(info, idx):
            'seeg' | 'bio' | 'chpi' | 'dipole' | 'gof' | 'ecog' | 'hbo' | 'hbr'
         Type of channel
     """
-    kind = info['chs'][idx]['kind']
-    if kind == FIFF.FIFFV_MEG_CH:
-        if info['chs'][idx]['unit'] == FIFF.FIFF_UNIT_T_M:
-            return 'grad'
-        elif info['chs'][idx]['unit'] == FIFF.FIFF_UNIT_T:
-            return 'mag'
-    elif kind == FIFF.FIFFV_REF_MEG_CH:
-        return 'ref_meg'
-    elif kind == FIFF.FIFFV_EEG_CH:
-        return 'eeg'
-    elif kind == FIFF.FIFFV_STIM_CH:
-        return 'stim'
-    elif kind == FIFF.FIFFV_EOG_CH:
-        return 'eog'
-    elif kind == FIFF.FIFFV_EMG_CH:
-        return 'emg'
-    elif kind == FIFF.FIFFV_ECG_CH:
-        return 'ecg'
-    elif kind == FIFF.FIFFV_RESP_CH:
-        return 'resp'
-    elif kind == FIFF.FIFFV_MISC_CH:
-        return 'misc'
-    elif kind == FIFF.FIFFV_EXCI_CH:
-        return 'exci'
-    elif kind == FIFF.FIFFV_IAS_CH:
-        return 'ias'
-    elif kind == FIFF.FIFFV_SYST_CH:
-        return 'syst'
-    elif kind == FIFF.FIFFV_SEEG_CH:
-        return 'seeg'
-    elif kind == FIFF.FIFFV_BIO_CH:
-        return 'bio'
-    elif kind in [FIFF.FIFFV_QUAT_0, FIFF.FIFFV_QUAT_1, FIFF.FIFFV_QUAT_2,
-                  FIFF.FIFFV_QUAT_3, FIFF.FIFFV_QUAT_4, FIFF.FIFFV_QUAT_5,
-                  FIFF.FIFFV_QUAT_6, FIFF.FIFFV_HPI_G, FIFF.FIFFV_HPI_ERR,
-                  FIFF.FIFFV_HPI_MOV]:
-        return 'chpi'  # channels relative to head position monitoring
-    elif kind == FIFF.FIFFV_DIPOLE_WAVE:
-        return 'dipole'
-    elif kind == FIFF.FIFFV_GOODNESS_FIT:
-        return 'gof'
-    elif kind == FIFF.FIFFV_ECOG_CH:
-        return 'ecog'
-    elif kind == FIFF.FIFFV_FNIRS_CH:
-        if info['chs'][idx]['coil_type'] == FIFF.FIFFV_COIL_FNIRS_HBO:
-            return 'hbo'
-        elif info['chs'][idx]['coil_type'] == FIFF.FIFFV_COIL_FNIRS_HBR:
-            return 'hbr'
+    ch = info['chs'][idx]
+
+    for t, rules in get_channel_types().items():
+        for key, vals in rules.items():
+            if isinstance(vals, int) and ch.get(key, None) != vals:
```

Comment:
I find the logic hard to parse. Can you simplify the testing?

Reply:
for instance changing this to `if ch.get(key, None) not in np.array(vals)` might avoid the need for the conditional (depending on how well separated the logic currently is)

Reply:
Yes, this gets rid of the condition. @larsoner I was wondering if this is possible with simple lists, by apparently there is no list constructor that does something similar to `np.array`, or is there?
@agramfort I've also added comments. Is the logic clearer now?